hatchways/team-gingerbread
# Notifications Model and Router

notifications model and router with corresponding routes for create, read, getAll, and getUnread

### What this PR does (required):
-fixes #83 
- creates Notifications model and router with routes
- routes: 
1. create new notification (/notifications/create, POST)
2. change read status (/notifications/read, POST)
3. get all notifications (/notifications/all, GET)
4. get unread notifications (/notifications/unread, GET)

### Any information needed to test this feature (required):
- run server in terminal
```
npm run dev
```
- link to MongoDB to see database changes
- open postman to test routes (start with step 1 as the database will need test notifications)

1. /notifications/create: in the request body, specify type, title, description, read, and date
2. /notifications/read: in the request body, specify the id of the notification you would like to change the read status to true
3. /notifications/all: sends all notifications
4. /notifications/unread: sends all notifications where "read" is equal to false

### Any issues with the current functionality (optional):
- the database stores all the notifications in a "notifications" collection. with multiple users, this database needs to be structured in a way to only have a user access their own notifications. I would appreciate some feedback on how I should implement the optimal design for this database
- if the database is redesigned to group notifications by the user, the user will have to pass their user id in each request, therefore each request will need to be a POST request. furthermore, the auth middleware will need to be implemented in the notifications router so that the user's credentials are verified
